crates/valence_server/src/status_effect.rs

```diff
@@ -0,0 +1,145 @@
+use bevy_app::prelude::*;
+use bevy_ecs::prelude::*;
+use valence_entity::active_status_effects::{ActiveStatusEffect, ActiveStatusEffects};
+use valence_entity::entity::Flags;
+use valence_entity::status_effects::StatusEffect;
+use valence_protocol::packets::play::{
+    entity_status_effect_s2c, EntityStatusEffectS2c, RemoveEntityStatusEffectS2c,
+};
+use valence_protocol::{VarInt, WritePacket};
+
+use crate::client::Client;
+use crate::EventLoopPostUpdate;
+
+pub struct StatusEffectPlugin;
+
+impl Plugin for StatusEffectPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_systems(
+            EventLoopPostUpdate,
+            (
+                remove_expired_status_effects,
+                update_active_status_effects,
+                add_status_effects,
+            ),
+        );
+    }
+}
+
+fn update_active_status_effects(mut query: Query<(&mut ActiveStatusEffects, Option<&mut Client>)>) {
+    for (mut active_status_effects, mut client) in query.iter_mut() {
+        for (_, effect) in active_status_effects.active_effects_mut().iter_mut() {
+            effect.decrement_duration();
+
+            // Like in vanilla, remind the client of the effect every 600 ticks
+            if effect.duration() > 0 && effect.duration() % 600 == 0 {
+                if let Some(ref mut client) = client {
+                    client.write_packet(&create_packet(effect));
+                }
+            }
+
+            /* TODO: The following things require to occasionally modify
+             * entity stuff:
+             * - regeneration
+             * - poison
+             * - wither
+             */
+        }
+    }
+}
+
+fn create_packet(effect: &ActiveStatusEffect) -> EntityStatusEffectS2c {
+    EntityStatusEffectS2c {
+        entity_id: VarInt(0),
```

Comment:
Why are you not taking entity id as a parameter? Does zero mean self? if I understand correctly every client in the same entity layer should receive the status effects of all entities in that layer. I understand if this is WIP disregard me.

Reply:
In other parts of the code across Valence, the ID of the player is 0. The server always seems to tell the client that it's 0. That's just what I'm following. If I make it be the entity id, then the client doesn't seem to think that it has any effect and thus nothing happens.